src/middlewared/middlewared/plugins/vm.py

```diff
@@ -1987,6 +2140,24 @@ def translate_device(dev):
         else:
             return False
 
+    @private
+    async def get_iommu_type(self):
+        IOMMU_TESTS = {'VT-d': {'cmd_args': ['/usr/sbin/acpidump', '-t'],
+                                'pattern': r'DMAR'},
+                       'amdvi': {'cmd_args': ['/sbin/sysctl', '-i', 'hw.vmm.amdvi.enable'],
+                                 'pattern': r'vmm\.amdvi\.enable: 1'}}
+
+        if self.iommu_type is None:
+            for key, value in IOMMU_TESTS.items():
+                sp = await run(*value['cmd_args'], check=False)
+                if sp.returncode:
+                    raise CallError(f'Failed to check support for iommu ({key}): {sp.stderr.decode()}')
+                else:
+                    if re.search(value['pattern'], sp.stdout.decode()):
```

Comment:
I am getting a decode error in my system for this
```
freenas# midclt call vm.device.get_iommu_type
'utf-8' codec can't decode byte 0xf1 in position 7552: invalid continuation byte
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/middlewared/main.py", line 135, in call_method
    result = await self.middleware._call(message['method'], serviceobj, methodobj, params, app=self,
  File "/usr/local/lib/python3.8/site-packages/middlewared/main.py", line 1158, in _call
    return await methodobj(*args)
  File "/usr/local/lib/python3.8/site-packages/middlewared/plugins/vm.py", line 2156, in get_iommu_type
    if re.search(value['pattern'], sp.stdout.decode()):
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xf1 in position 7552: invalid continuation byte
```

I think you should probably use `br'DMAR'` and not try to decode it. It should be safe I assume in this case

Reply:
@sonicaj, thank you for testing. Interesting and, although the output of acpidump is heavily dependant on system hw, somewhat unexpected. On my system, I don't get that error :-) I agree that using bytes instead of string is probably ok in this case. Will update accordingly. 
Trying to understand what caused the decode error on your system, would you mind sharing the output line with the offending bytes?